### PR TITLE
LogStream: Remove testing mode and float manipulation

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #4704

<s>Do not merge yet - I need a testsuite run first to fix up tests.</s>

### CODE REVIEW

**Comment by tjhei** [comment]
:+1: Good to merge when tests are fixed.

**Comment by tamiko** [comment]
The PR now passes the testsuite. I will rewrite `trilinos/precondition` to make it more robust (this is somewhat unrelated to this PR).

File: cmake/scripts/run_test.sh [hunk 1]
```diff
@@ -105,7 +105,7 @@ case $STAGE in
       #
       case "${NUMDIFF_EXECUTABLE}" in
         *numdiff*)
-          "${NUMDIFF_EXECUTABLE}" -a 1e-6 -r 1e-8 -s ' \t\n:<>=,;' \
+          "${NUMDIFF_EXECUTABLE}" -a 1e-6 -r 1e-8 -s ' \t\n=,:;<>[](){}^' \
```
Comment (bangerth) [inline cmake/scripts/run_test.sh]: ugh :-)

Comment (bangerth) [inline cmake/scripts/run_test.sh]: Could we have a comment that describes what exactly this does?

**Comment by tamiko** [comment]
@bangerth I added a small comment.

Testsuite (regression tester) result: http://cdash.kyomu.43-1.org/buildSummary.php?buildid=10038

**Comment by bangerth** [comment]
Great, thanks!

**Pull request merged**